Mathlib/Topology/CWComplex/Basic.lean

```diff
@@ -0,0 +1,684 @@
+    MapsTo (map n i) (sphere 0 1) (⋃ (m < n) (j ∈ I m), map m j '' closedBall 0 1) := by
+  have := RelCWComplex.mapsto n i
+  simp_rw [empty_union] at this
+  exact this
+
+lemma RelCWComplex.pairwiseDisjoint [RelCWComplex C D] :
+    (univ : Set (Σ n, cell C n)).PairwiseDisjoint (fun ni ↦ openCell ni.1 ni.2) :=
+  RelCWComplex.pairwiseDisjoint'
+
+lemma RelCWComplex.disjointBase [RelCWComplex C D] (n : ℕ) (i : cell C n) :
+    Disjoint (openCell n i) D :=
+  RelCWComplex.disjointBase' n i
+
+lemma RelCWComplex.disjoint_openCell_of_ne [RelCWComplex C D] {n m : ℕ} {i : cell C n}
+    {j : cell C m} (ne : (⟨n, i⟩ : Σ n, cell C n) ≠ ⟨m, j⟩) : openCell n i ∩ openCell m j = ∅ := by
+  have := pairwiseDisjoint (C := C) (D := D)
+  simp only [PairwiseDisjoint, Set.Pairwise, Function.onFun, disjoint_iff_inter_eq_empty] at this
+  exact this (mem_univ _) (mem_univ _) ne
+
+lemma RelCWComplex.cellFrontier_subset_base_union_finite_closedCell [RelCWComplex C D]
+    (n : ℕ) (i : cell C n) : ∃ I : Π m, Finset (cell C m), cellFrontier n i ⊆
+    D ∪ ⋃ (m < n) (j ∈ I m), closedCell m j := by
+  rcases mapsto n i with ⟨I, hI⟩
+  use I
+  rw [mapsTo'] at hI
+  exact hI
+
+lemma ClasCWComplex.cellFrontier_subset_finite_closedCell [ClasCWComplex C] (n : ℕ) (i : cell C n) :
+    ∃ I : Π m, Finset (cell C m), cellFrontier n i ⊆ ⋃ (m < n) (j ∈ I m), closedCell m j := by
+  rcases RelCWComplex.mapsto n i with ⟨I, hI⟩
+  use I
+  rw [mapsTo', empty_union] at hI
+  exact hI
+
+lemma RelCWComplex.union [RelCWComplex C D] : D ∪ ⋃ (n : ℕ) (j : cell C n), closedCell n j = C :=
+  RelCWComplex.union'
+
+lemma ClasCWComplex.union [ClasCWComplex C] : ⋃ (n : ℕ) (j : cell C n), closedCell n j = C := by
+  have := RelCWComplex.union' (C := C) (D := ∅)
+  rw [empty_union] at this
+  exact this
+
+lemma RelCWComplex.openCell_subset_closedCell [RelCWComplex C D] (n : ℕ) (i : cell C n) :
+    openCell n i ⊆ closedCell n i := image_mono Metric.ball_subset_closedBall
+
+lemma RelCWComplex.cellFrontier_subset_closedCell [RelCWComplex C D] (n : ℕ) (i : cell C n) :
+    cellFrontier n i ⊆ closedCell n i := image_mono Metric.sphere_subset_closedBall
+
+lemma RelCWComplex.cellFrontier_union_openCell_eq_closedCell [RelCWComplex C D] (n : ℕ)
+    (i : cell C n) : cellFrontier n i ∪ openCell n i = closedCell n i := by
+  rw [cellFrontier, openCell, closedCell, ← image_union]
+  congrm map n i '' ?_
+  exact sphere_union_ball
+
+lemma RelCWComplex.map_zero_mem_openCell [RelCWComplex C D] (n : ℕ) (i : cell C n) :
+    map n i 0 ∈ openCell n i := by
+  apply mem_image_of_mem
+  simp only [mem_ball, dist_self, zero_lt_one]
+
+lemma RelCWComplex.map_zero_mem_closedCell [RelCWComplex C D] (n : ℕ) (i : cell C n) :
+    map n i 0 ∈ closedCell n i :=
+  openCell_subset_closedCell _ _ (map_zero_mem_openCell _ _)
+
+/-- A non-standard definition of the `n`-th level of a CW complex for `n ∈ ℕ ∪ ∞` useful for
+  induction. The standard `level` is defined in terms of levelaux. `levelaux` is preferred in
```

Comment:
Could you explain in more detail why you need this distinction between levelaux and level? 

Could we give it a more intuitive name? e.g. `below` or `cellsLT` or `skeletonLT`, etc?